adapters/testutils/synctransaction_test_utils.go

```diff
@@ -0,0 +1,584 @@
+		t,
+		&consensusDeclareTransaction,
+		network,
+	)
+	return b.ToCore(&consensusDeclareTransaction, nil, nil),
+		b.ToP2PDeclareV2(&p2pTransaction, p2pHash)
+}
+
+func (b *SyncTransactionBuilder[C, P]) GetTestDeclareV3Transaction(
+	t *testing.T,
+	network *utils.Network,
+) (C, P) {
+	t.Helper()
+	classHash, cairo1Class := getSampleClass(t)
+	senderAddress, senderAddressBytes := getRandomFelt(t)
+	transactionSignature, transactionSignatureBytes := getRandomFeltSlice(t)
+	nonce, nonceBytes := getRandomFelt(t)
+	version := new(core.TransactionVersion).SetUint64(3)
+	resourceBounds, p2pResourceBounds := getRandomResourceBounds(t)
+	tip := rand.Uint64()
+	paymasterData, paymasterDataBytes := getRandomFeltSlice(t)
+	accountDeploymentData, accountDeploymentDataBytes := getRandomFeltSlice(t)
+
+	p2pTransaction := synctransaction.TransactionInBlock_DeclareV3WithoutClass{
+		Common: &transaction.DeclareV3Common{
+			Sender: &common.Address{Elements: senderAddressBytes},
+			Signature: &transaction.AccountSignature{
+				Parts: toFelt252Slice(transactionSignatureBytes),
+			},
+			Nonce:                     &common.Felt252{Elements: nonceBytes},
+			CompiledClassHash:         core2p2p.AdaptHash(cairo1Class.Compiled.Hash()),
+			ResourceBounds:            p2pResourceBounds,
+			Tip:                       tip,
+			PaymasterData:             toFelt252Slice(paymasterDataBytes),
+			AccountDeploymentData:     toFelt252Slice(accountDeploymentDataBytes),
+			NonceDataAvailabilityMode: common.VolitionDomain_L2,
+			FeeDataAvailabilityMode:   common.VolitionDomain_L2,
+		},
+		ClassHash: core2p2p.AdaptHash(&classHash),
+	}
+
+	consensusDeclareTransaction := core.DeclareTransaction{
+		TransactionHash:       nil, // this field is populated later
+		ClassHash:             &classHash,
+		SenderAddress:         &senderAddress,
+		MaxFee:                nil, // this field is not available on v3
+		TransactionSignature:  transactionSignature,
+		Nonce:                 &nonce,
+		Version:               version,
+		CompiledClassHash:     cairo1Class.Compiled.Hash(),
+		ResourceBounds:        resourceBounds,
+		Tip:                   tip,
+		PaymasterData:         paymasterData,
+		AccountDeploymentData: accountDeploymentData,
+		NonceDAMode:           core.DAModeL2,
+		FeeDAMode:             core.DAModeL2,
+	}
+
+	var p2pHash *common.Hash
+	consensusDeclareTransaction.TransactionHash, p2pHash = getTransactionHash(
+		t,
+		&consensusDeclareTransaction,
+		network,
+	)
+	return b.ToCore(&consensusDeclareTransaction, nil, nil),
+		b.ToP2PDeclareV3(&p2pTransaction, p2pHash)
+}
+
+func (b *SyncTransactionBuilder[C, P]) GetTestDeployTransactionV0(
+	t *testing.T,
+	network *utils.Network,
+) (C, P) {
+	t.Helper()
+	contractAddressSalt, contractAddressSaltBytes := getRandomFelt(t)
+	classHash, _ := getRandomFelt(t)
+	constructorCallData, _ := getRandomFeltSlice(t)
+	contractAddress := core.ContractAddress(
+		&felt.Zero, &classHash,
```

Comment:
nit, same for the other functions:
```suggestion
		&felt.Zero,
		&classHash,
```